util/gnmi.go

```diff
@@ -134,6 +134,41 @@ func FindPathElemPrefix(paths []*gpb.Path) *gpb.Path {
 	}
 }
 
+// pathElemMatchesQuery returns true if the given concrete (no wildcard)
+// PathElem is a possible gNMI telemetry name match for the reference PathElem.
+// In particular, it matches wildcards for names and list keys, but *not* "...".
+// See https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-path-conventions.md
+func pathElemMatchesQuery(elem, refElem *gpb.PathElem) bool {
+	if elem == nil || refElem == nil {
+		return elem == nil && refElem == nil
+	}
+
+	if refElem.Name != "*" && elem.Name != refElem.Name {
+		return false
+	}
+
+	for k, ref := range refElem.Key {
+		if v, ok := elem.Key[k]; !ok || (ref != "*" && v != ref) {
```

Comment:
If `ref == "*"` then you don't need to check that `v != ref` right? Because even if `v == ref` then if `ref == *` we'll return true.